Upstream-ready timberland_upstream+timberland_1.0_final

trevor-cockrell opened the pull request:
Combine `timberland_upstream` and `timberland_1.0_final` into a singular, upstream-ready branch.

Rebased:
- `timberland_upstream`
  - rebased to resolve items mentioned below
- `timberland_1.0_final` (based on https://github.com/timberland-sig/edk2/pull/22) 
  - rebased to resolve below items as well as modularize for upstream EDK2-readiness
 
Resolved:
- PatchCheck issues
  - convert line endings (LF->CRLF)
  - minor commit message formatting
- uncrustify formatting on all touched files
- Coverity errors introduced by base package

PatchCheck results:
```
$ python ./BaseTools/Scripts/PatchCheck.py fff6d81270..HEAD --oneline

NetworkPkg/Library/DxeSpdkLib: Basic NvmeOf IPv6 support
628aeefe49 ok
ShellPkg/Library/UefiShellCommandLib: NvmeOf device path
cc3cb4e243 ok
NetworkPkg/NvmeOfDxe: DHCPv6 option 59 support
c1ac9bed4b ok
NetworkPkg: build VConfig.efi if NETWORK_VLAN_ENABLE is set
3a746db2e1 ok
NetworkPkg: Enable NvmeOfCli
a44f01de83 ok
EmulatorPkg: Enable NvmeOfCli
276302c3a5 ok
OvmfPkg: Enable NvmeOfCli
33f6644473 ok
NetworkPkg/NvmeOfDxe: Populate NBFT table for failure cases.
b1f5aaad4f ok
NetworkPkg/Application: Introduce NvmeOfCli application
242e476411 ok
NetworkPkg/NvmeOfDxe: NvmeOfPassthru socket changes
0922c3d4c1 ok
MdePkg/Include/Protocol: NvmeOfPassthru socket changes
aadc6d99ca ok
MdeModulePkg/Library/UefiBootManagerLib: NvmeOfPassthru
774b0888c7 ok
NetworkPkg/NvmeOfDxe: NvmeOfCliInterface integration
9891566555 ok
MdePkg/Include/Protocol: NvmeOfPassthru prototype additions
c760e47623 ok
EmulatorPkg: Add NETWORK_NVMEOF_ENABLE define to DSC file.
6810e2407f ok
OvmfPkg: Enable NVMe-over-Fabrics boot feature.
f67395152b ok
NetworkPkg: Introduce NvmeOfDxe driver.
8a1b427777 ok
MdeModulePkg: Notify BeforeExitBootServices in CoreExitBootServices
3fac9c9c52 ok
MdePkg: Add EFI_EVENT_BEFORE_EXIT_BOOT_SERVICES_GUID
5be30ef82c ok
NetworkPkg: Introduce DxeSpdkLib (SPDK port to EDK2).
2aca375061 ok
NetworkPkg/TcpDxe: Add IP_IO Tx callback for handling Tx errors.
2b730d151c ok
MdeModulePkg/UefiBootManagerLib: Add boot desc for NVMe-oF device paths.
2bf168d047 ok
MdePkg: Added EDKII_NVMEOF_PASSTHRU_PROTOCOL definition and GUID.
225c2a1aa5 ok
BaseTools/DevicePath: Added NVMe-oF Device Path node and parse function.
c24b4f4e7a ok
MdePkg/DiskInfo: Added NVMe-oF interface GUID.
6ae56f618e ok
MdePkg/UefiDevicePathLib: Added NVMe-oF node-string parsing.
3075e53992 ok
MdePkg/DevicePath: Added NVMe-oF Device Path node.
1c4b208001 ok
MdePkg/Acpi: Add NBFT table definition.
6b26c1b0ea ok
Added readme for NVMe-over-Fabrics boot prototype upstreaming.
fdda1d24e3 ok
```

deeglaze committed:
MdePkg: Add EFI_EVENT_BEFORE_EXIT_BOOT_SERVICES_GUID

Event group as defined in UEFI standard v2.9.

Cc: Ard Biescheuvel <ardb@kernel.org>
Cc: "Min M. Xu" <min.m.xu@intel.com>
Cc: Gerd Hoffmann <kraxel@redhat.com>
Cc: James Bottomley <jejb@linux.ibm.com>
Cc: Tom Lendacky <Thomas.Lendacky@amd.com>
Cc: Jiewen Yao <jiewen.yao@intel.com>
Cc: Erdem Aktas <erdemaktas@google.com>

Acked-by: Jiewen Yao <Jiewen.yao@intel.com>
Reviewed-by: Michael D Kinney <michael.d.kinney@intel.com>
Signed-off-by: Dionna Glaze <dionnaglaze@google.com>
Message-Id: <20221108164616.3251967-3-dionnaglaze@google.com>

deeglaze committed:
MdeModulePkg: Notify BeforeExitBootServices in CoreExitBootServices

Location of notification is has been specified in UEFI v2.9.

Cc: Gerd Hoffmann <kraxel@redhat.com>
Cc: James Bottomley <jejb@linux.ibm.com>
Cc: Jiewen Yao <jiewen.yao@intel.com>
Cc: Tom Lendacky <thomas.lendacky@amd.com>
Cc: Ard Biesheuvel <ardb@kernel.org>
Cc: "Min M. Xu" <min.m.xu@intel.com>
Cc: Andrew Fish <afish@apple.com>
Cc: "Michael D. Kinney" <michael.d.kinney@intel.com>
Cc: Ray Ni <ray.ni@intel.com>

Acked-by: Jiewen Yao <Jiewen.yao@intel.com>
Reviewed-by: Michael D Kinney <michael.d.kinney@intel.com>
Signed-off-by: Dionna Glaze <dionnaglaze@google.com>
Message-Id: <20221108164616.3251967-4-dionnaglaze@google.com>

Douglas-Farley committed:
Added readme for NVMe-over-Fabrics boot prototype upstreaming.

Outline the project, its goals, maintainers and contributors.

Signed-off-by: Doug Farley <Douglas_Farley@dell.com>
Cc: Richelle Ahlvers <richelle.ahlvers@intel.com>
Cc: Maciej Rabeda <maciej.rabeda@linux.intel.com>
Change-Id: I13aa3e90793e5685cfddd730d060051b38fc00c3

Commit:
MdePkg/Acpi: Add NBFT table definition.

NBFT ACPI table definition was introduced in NVM Express Boot
Specification 1.0. It is a handoff information structure exchanged
between UEFI environment and operating system.

Signed-off-by: Doug Farley <Douglas_Farley@dell.com>
Co-authored-by: Amit Jain <amit_jain9@dell.com>

Commit:
MdePkg/DevicePath: Added NVMe-oF Device Path node.

Definition of this node was introduced in UEFI specification v2.10.

Signed-off-by: Doug Farley <Douglas_Farley@dell.com>
Co-authored-by: Amit Jain <amit_jain9@dell.com>
Co-authored-by: Lenny Szubowicz <lszubowi@redhat.com>

Commit:
MdePkg/UefiDevicePathLib: Added NVMe-oF node-string parsing.

Signed-off-by: Doug Farley <Douglas_Farley@dell.com>
Co-authored-by: Amit Jain <amit_jain9@dell.com>
Co-authored-by: Maciej Rabeda <maciej.rabeda@intel.com>

Commit:
MdePkg/DiskInfo: Added NVMe-oF interface GUID.

TODO: UEFI PI specification update.

Signed-off-by: Doug Farley <Douglas_Farley@dell.com>

Commit:
BaseTools/DevicePath: Added NVMe-oF Device Path node and parse function.

Definition of this node was introduced in UEFI specification v2.10.

Signed-off-by: Doug Farley <Douglas_Farley@dell.com>
Co-authored-by: Lenny Szubowicz <lszubowi@redhat.com>
Co-authored-by: Maciej Rabeda <maciej.rabeda@intel.com>

mrabeda committed:
MdePkg: Added EDKII_NVMEOF_PASSTHRU_PROTOCOL definition and GUID.

This protocol is required (at minimum) by Boot Manager libraries to
obtain boot description for a handle supported by NVMe-over-Fabrics boot
driver(s).

Signed-off-by: Doug Farley <Douglas_Farley@dell.com>
Co-authored-by: Amit Jain <amit_jain9@dell.com>
Co-authored-by: Hrishikesh Gokhale <Hrishikesh_Gokhale@dell.com>

Commit:
MdeModulePkg/UefiBootManagerLib: Add boot desc for NVMe-oF device paths.

Signed-off-by: Doug Farley <Douglas_Farley@dell.com>
Co-authored-by: Hrishikesh Gokhale <Hrishikesh_Gokhale@dell.com>
Co-authored-by: Maciej Rabeda <maciej.rabeda@intel.com>

mrabeda committed:
NetworkPkg/TcpDxe: Add IP_IO Tx callback for handling Tx errors.

This allows to convey errors that happened at lower layers of network
stack to upper layers.

Signed-off-by: Doug Farley <Douglas_Farley@dell.com>

Commit:
NetworkPkg: Introduce DxeSpdkLib (SPDK port to EDK2).

SPDK submodule currently points to a Timberland fork of SPDK which
contains necessary changes for it to build with MSVC compiler.

Signed-off-by: Doug Farley <Douglas_Farley@dell.com>
Co-authored-by: Murali Manohar Shanm <Murali_Manohar_Shanm@Dell.com>
Co-authored-by: Naoyuki Mori <naoyuki.mori@intel.com>
Co-authored-by: Maciej Rabeda <maciej.rabeda@intel.com>

Commit:
NetworkPkg: Introduce NvmeOfDxe driver.

This driver allows for connecting to remote NVMe namespaces over TCP
fabric and expose them as block devices upwards the software stack.

Signed-off-by: Doug Farley <Douglas_Farley@dell.com>
Co-authored-by: Amit Jain <amit_jain9@dell.com>
Co-authored-by: Hrishikesh Gokhale <Hrishikesh_Gokhale@dell.com>
Co-authored-by: Maciej Rabeda <maciej.rabeda@intel.com>
Co-authored-by: Naoyuki Mori <naoyuki.mori@intel.com>
Co-authored-by: Martin Wilck <mwilck@suse.com>

Commit:
OvmfPkg: Enable NVMe-over-Fabrics boot feature.

Signed-off-by: Doug Farley <Douglas_Farley@dell.com>
Co-authored-by: Amit Jain <amit_jain9@dell.com>

Commit:
EmulatorPkg: Add NETWORK_NVMEOF_ENABLE define to DSC file.

Signed-off-by: Doug Farley <Douglas_Farley@dell.com>

trevor-cockrell committed:
NetworkPkg: Add Pcd for maximum NVMe-oF attempts

Add PcdMaxNvmeOfAttemptNumber to NetworkPkg as the maximum number of
NVMe-over-fabrics attempts.

Signed-off-by: Trevor cockrell <trevor_cockrell@dell.com>

trevor-cockrell committed:
NetworkPkg/Include: Add GUID gNvmeOfConfigGuid

GUID gNvmeOfConfigGuid added for NvmeOfDxe HII configuration forms.

Signed-off-by: Trevor cockrell <trevor_cockrell@dell.com>

trevor-cockrell committed:
NetworkPkg/NvmeOfDxe: NVMe-over-Fabrics configuration forms

Add support for modification of NVMe-oF connection attempts in the
NVMe-oF configuration submenu of the HII.
Update NvmeOfDxe readme with configuration form details.

Signed-off-by: Trevor cockrell <trevor_cockrell@dell.com>

Commit:
OvmfPkg: Enable NvmeOfCli

ShellPkg support for NvmeOfCli in OvmfPkg

Signed-off-by: Trevor Cockrell <trevor_cockrell@dell.com>

Commit:
EmulatorPkg: Enable NvmeOfCli

ShellPkg support for NvmeOfCli

Signed-off-by: Trevor Cockrell <trevor_cockrell@dell.com>

Commit:
NetworkPkg: Enable NvmeOfCli

Signed-off-by: Trevor Cockrell <trevor_cockrell@dell.com>

mwilck committed:
NetworkPkg: build VConfig.efi if NETWORK_VLAN_ENABLE is set

Signed-off-by: Trevor Cockrell <trevor_cockrell@dell.com>

Commit:
NetworkPkg/NvmeOfDxe: DHCPv6 option 59 support

Fixes DHCPv6 root-path support to use the correct DHCP Option (59)

Signed-off-by: Trevor Cockrell <trevor_cockrell@dell.com>

Commit:
ShellPkg/Library/UefiShellCommandLib: NvmeOf device path

Add NvmeOf DevicePath support.

Signed-off-by: Trevor Cockrell <trevor_cockrell@dell.com>

Commit:
MdePkg/Include/Protocol: NvmeOfPassthru prototype additions

Added protocol structure definition compliant with all other protocol
definitions in EDK2

Signed-off-by: Trevor Cockrell <trevor_cockrell@dell.com>

Commit:
NetworkPkg/NvmeOfDxe: NvmeOfCliInterface integration

Add and integrate NvmeOfCliInterface changes utilizing NvmeOfPassthru
protocol.

Signed-off-by: Trevor Cockrell <trevor_cockrell@dell.com>

Commit:
MdeModulePkg/Library/UefiBootManagerLib: NvmeOfPassthru

Align NvmeOf device description with updated NvmeOfPassthru protocol.

Signed-off-by: Trevor Cockrell <trevor_cockrell@dell.com>

Commit:
MdePkg/Include/Protocol: NvmeOfPassthru socket changes

Port socket context changes to NvmeOfPassthru for NvmeOfCli

Signed-off-by: Trevor Cockrell <trevor_cockrell@dell.com>

Commit:
NetworkPkg/NvmeOfDxe: NvmeOfPassthru socket changes

- Used Attempt Entry wrapper structure to store the socket context to be
  used while socket connections.
- Code changes to free the context with attempt entry on cli disconnect.

Signed-off-by: Trevor Cockrell <trevor_cockrell@dell.com>

Commit:
NetworkPkg/Application: Introduce NvmeOfCli application

Signed-off-by: Trevor Cockrell <trevor_cockrell@dell.com>
Co-authored-by: Hrishikesh Gokhale <Hrishikesh_Gokhale@dell.com>

Commit:
NetworkPkg/NvmeOfDxe: Populate NBFT table for failure cases.

This change is to populate the details of failed connections to NBFT.
This applies to such subsystems, for which attempt has been configured
but the NVMe-oF driver was unable to create a successful connection to
them. This change will enable the booting OS to get the details of such
subsystems from NBFT and retry the connection to proceed with booting
from them.

Signed-off-by: Trevor Cockrell <trevor_cockrell@dell.com>

naoyuki-mori committed:
NetworkPkg/Library/DxeSpdkLib: Basic NvmeOf IPv6 support

This patch brings in basic IPv6 Support for the NvmeOfDxe driver via the
DxeSpdkLib library.

Changes:
* Library/DxeSpdkLib: Added IPv6 support into non-blocking rx edk_sock
  shim
* NetworkPkg DxeSpdkLib edk_sock.c: Coding standard fixes.
* Library/DxeSpdkLib: Refactor to unify expression of edk_sock

Known Issues:
* SLAAC and RA still have some issues still being worked - specifically
  both environments may sometimes see a hang
* Configuration of IPV6 should be done via Hii IPv6 config and not
  (always) NvmeOfCli

Signed-off-by: Trevor Cockrell <trevor_cockrell@dell.com>

trevor-cockrell commented:
HII feature from https://github.com/timberland-sig/edk2/pull/24 has been modularized and added as 3 commits, divided by module:

- eccf7ac: NetworkPkg
  - Addition of PcdMaxNvmeOfAttemptNumber
- 68a2130: NetworkPkg/Include
  - Addition of gNvmeOfConfigGuid
- c0182aa: NetworkPkg/NvmeOfDxe
  - Add HII/configuration forms
  - Add HII details in NvmeOfDxe README

PatchCheck output:
``` 
NetworkPkg/NvmeOfDxe: NVMe-over-Fabrics configuration forms
c0182aa680 ok
NetworkPkg/Include: Add GUID gNvmeOfConfigGuid
68a2130cab ok
NetworkPkg: Add Pcd for maximum NVMe-oF attempts
eccf7ac90b ok
```

tbzatek commented:
I'm having hard time getting this branch compiled, while other branches do compile fine for me:

```
$ build -t GCC5 -a X64 -p OvmfPkg/OvmfPkgX64.dsc
Build environment: Linux-6.2.2-zen-g497366a83d4a-x86_64-Intel-R-_Core-TM-_i5-8500_CPU_@_3.00GHz-with-glibc2.38
Build start time: 17:39:56, Oct.11 2023

WORKSPACE        = /tmp/edk2
EDK_TOOLS_PATH   = /tmp/edk2/BaseTools
CONF_PATH        = /tmp/edk2/Conf
PYTHON_COMMAND   = /usr/bin/python3



Processing meta-data .Architecture(s)  = X64
Build target     = DEBUG
Toolchain        = GCC5

Active Platform          = /tmp/edk2/OvmfPkg/OvmfPkgX64.dsc
... done!
```
...
```
Building ... /tmp/edk2/NetworkPkg/IScsiDxe/IScsiDxe.inf [X64]
"gcc" -MMD -MF /tmp/edk2/Build/OvmfX64/DEBUG_GCC5/X64/NetworkPkg/IScsiDxe/IScsiDxe/OUTPUT/AutoGen.obj.deps -g -Os -fshort-wchar -fno-builtin -fno-strict-aliasing -Wall -Werror -Wno-array-bounds -include AutoGen.h -fno-common -ffunction-sections -fdata-sections -DSTRING_ARRAY_NAME=IScsiDxeStrings -m64 -fno-stack-protector "-DEFIAPI=__attribute__((ms_abi))" -maccumulate-outgoing-args -mno-red-zone -Wno-address -mcmodel=small -fpie -fno-asynchronous-unwind-tables -Wno-address -flto -DUSING_LTO -Os -mno-mmx -mno-sse -D DISABLE_NEW_DEPRECATED_INTERFACES -D TDX_GUEST_SUPPORTED -D ENABLE_MD5_DEPRECATED_INTERFACES -c -o /tmp/edk2/Build/OvmfX64/DEBUG_GCC5/X64/NetworkPkg/IScsiDxe/IScsiDxe/OUTPUT/./AutoGen.obj -I/tmp/edk2/NetworkPkg/IScsiDxe -I/tmp/edk2/Build/OvmfX64/DEBUG_GCC5/X64/NetworkPkg/IScsiDxe/IScsiDxe/DEBUG -I/tmp/edk2/MdePkg -I/tmp/edk2/MdePkg/Include -I/tmp/edk2/MdePkg/Test/UnitTest/Include -I/tmp/edk2/MdePkg/Include/X64 -I/tmp/edk2/MdeModulePkg -I/tmp/edk2/MdeModulePkg/Include -I/tmp/edk2/CryptoPkg -I/tmp/edk2/CryptoPkg/Include -I/tmp/edk2/NetworkPkg -I/tmp/edk2/NetworkPkg/Include -I/tmp/edk2/NetworkPkg/Library/DxeSpdkLib/SpdkShim -I/tmp/edk2/NetworkPkg/Library/DxeSpdkLib/SpdkShim/sys -I/tmp/edk2/NetworkPkg/Library/DxeSpdkLib/SpdkShim/spdk -I/tmp/edk2/NetworkPkg/Library/DxeSpdkLib/spdk/include -I/tmp/edk2/NetworkPkg/Library/DxeSpdkLib/spdk/lib/nvme /tmp/edk2/Build/OvmfX64/DEBUG_GCC5/X64/NetworkPkg/IScsiDxe/IScsiDxe/DEBUG/AutoGen.c
In file included from /tmp/edk2/NetworkPkg/NvmeOfDxe/NvmeOfDriver.c:20:
/tmp/edk2/NetworkPkg/NvmeOfDxe/NvmeOfDriver.c: In function ‘NvmeofClearNbftData’:
/tmp/edk2/NetworkPkg/NvmeOfDxe/NvmeOfDriver.c:835:46: error: ‘fail_conn’ undeclared (first use in this function)
  835 |   NET_LIST_FOR_EACH_SAFE (Entry, NextEntry, &fail_conn) {
      |                                              ^~~~~~~~~
/tmp/edk2/NetworkPkg/Include/Library/NetLib.h:617:16: note: in definition of macro ‘NET_LIST_FOR_EACH_SAFE’
  617 |   for(Entry = (ListHead)->ForwardLink, NextEntry = Entry->ForwardLink; \
      |                ^~~~~~~~
/tmp/edk2/NetworkPkg/NvmeOfDxe/NvmeOfDriver.c:835:46: note: each undeclared identifier is reported only once for each function it appears in
  835 |   NET_LIST_FOR_EACH_SAFE (Entry, NextEntry, &fail_conn) {
      |                                              ^~~~~~~~~
/tmp/edk2/NetworkPkg/Include/Library/NetLib.h:617:16: note: in definition of macro ‘NET_LIST_FOR_EACH_SAFE’
  617 |   for(Entry = (ListHead)->ForwardLink, NextEntry = Entry->ForwardLink; \
      |                ^~~~~~~~
/tmp/edk2/NetworkPkg/Include/Library/NetLib.h:617:38: error: left-hand operand of comma expression has no effect [-Werror=unused-value]
  617 |   for(Entry = (ListHead)->ForwardLink, NextEntry = Entry->ForwardLink; \
      |                                      ^
/tmp/edk2/NetworkPkg/NvmeOfDxe/NvmeOfDriver.c:835:3: note: in expansion of macro ‘NET_LIST_FOR_EACH_SAFE’
  835 |   NET_LIST_FOR_EACH_SAFE (Entry, NextEntry, &fail_conn) {
      |   ^~~~~~~~~~~~~~~~~~~~~~
In file included from /tmp/edk2/MdePkg/Include/Uefi/UefiBaseType.h:16,
                 from /tmp/edk2/MdePkg/Include/Uefi.h:17,
                 from /tmp/edk2/Build/OvmfX64/DEBUG_GCC5/X64/NetworkPkg/NvmeOfDxe/NvmeOfDxe/DEBUG/AutoGen.h:16,
                 from <command-line>:
/tmp/edk2/MdePkg/Include/Base.h:754:42: error: invalid use of undefined type ‘struct spdk_nvme_fail_trid’
  754 | #define OFFSET_OF(TYPE, Field)  ((UINTN) __builtin_offsetof(TYPE, Field))
      |                                          ^~~~~~~~~~~~~~~~~~
/tmp/edk2/MdePkg/Include/Base.h:838:71: note: in expansion of macro ‘OFFSET_OF’
  838 | #define BASE_CR(Record, TYPE, Field)  ((TYPE *) ((CHAR8 *) (Record) - OFFSET_OF (TYPE, Field)))
      |                                                                       ^~~~~~~~~
/tmp/edk2/NetworkPkg/Include/Library/NetLib.h:599:11: note: in expansion of macro ‘BASE_CR’
  599 |           BASE_CR(Entry, Type, Field)
      |           ^~~~~~~
/tmp/edk2/NetworkPkg/NvmeOfDxe/NvmeOfDriver.c:836:20: note: in expansion of macro ‘NET_LIST_USER_STRUCT’
  836 |     FailTridInfo = NET_LIST_USER_STRUCT (Entry, struct spdk_nvme_fail_trid, link);
      |                    ^~~~~~~~~~~~~~~~~~~~
/tmp/edk2/NetworkPkg/NvmeOfDxe/NvmeOfDriver.c: In function ‘NvmeOfDriverEntry’:
/tmp/edk2/NetworkPkg/NvmeOfDxe/NvmeOfDriver.c:2120:24: error: ‘fail_conn’ undeclared (first use in this function)
 2120 |   InitializeListHead (&fail_conn);
      |                        ^~~~~~~~~
cc1: all warnings being treated as errors
"gcc" -MMD -MF /tmp/edk2/Build/OvmfX64/DEBUG_GCC5/X64/NetworkPkg/IScsiDxe/IScsiDxe/OUTPUT/ComponentName.obj.deps -g -Os -fshort-wchar -fno-builtin -fno-strict-aliasing -Wall -Werror -Wno-array-bounds -include AutoGen.h -fno-common -ffunction-sections -fdata-sections -DSTRING_ARRAY_NAME=IScsiDxeStrings -m64 -fno-stack-protector "-DEFIAPI=__attribute__((ms_abi))" -maccumulate-outgoing-args -mno-red-zone -Wno-address -mcmodel=small -fpie -fno-asynchronous-unwind-tables -Wno-address -flto -DUSING_LTO -Os -mno-mmx -mno-sse -D DISABLE_NEW_DEPRECATED_INTERFACES -D TDX_GUEST_SUPPORTED -D ENABLE_MD5_DEPRECATED_INTERFACES -c -o /tmp/edk2/Build/OvmfX64/DEBUG_GCC5/X64/NetworkPkg/IScsiDxe/IScsiDxe/OUTPUT/./ComponentName.obj -I/tmp/edk2/NetworkPkg/IScsiDxe -I/tmp/edk2/Build/OvmfX64/DEBUG_GCC5/X64/NetworkPkg/IScsiDxe/IScsiDxe/DEBUG -I/tmp/edk2/MdePkg -I/tmp/edk2/MdePkg/Include -I/tmp/edk2/MdePkg/Test/UnitTest/Include -I/tmp/edk2/MdePkg/Include/X64 -I/tmp/edk2/MdeModulePkg -I/tmp/edk2/MdeModulePkg/Include -I/tmp/edk2/CryptoPkg -I/tmp/edk2/CryptoPkg/Include -I/tmp/edk2/NetworkPkg -I/tmp/edk2/NetworkPkg/Include -I/tmp/edk2/NetworkPkg/Library/DxeSpdkLib/SpdkShim -I/tmp/edk2/NetworkPkg/Library/DxeSpdkLib/SpdkShim/sys -I/tmp/edk2/NetworkPkg/Library/DxeSpdkLib/SpdkShim/spdk -I/tmp/edk2/NetworkPkg/Library/DxeSpdkLib/spdk/include -I/tmp/edk2/NetworkPkg/Library/DxeSpdkLib/spdk/lib/nvme /tmp/edk2/NetworkPkg/IScsiDxe/ComponentName.c
make: *** [GNUmakefile:381: /tmp/edk2/Build/OvmfX64/DEBUG_GCC5/X64/NetworkPkg/NvmeOfDxe/NvmeOfDxe/OUTPUT/NvmeOfDriver.obj] Error 1
```

(gcc version 13.2.1 20230826)

trevor-cockrell commented:
> I'm having hard time getting this branch compiled, while other branches do compile fine for me:

This might be an issue with the cached spdk submodule not updating from remote.
Can you give it a shot after reinitializing spdk with `--remote` after checking out? 
` git submodule update --remote NetworkPkg/Library/DxeSpdkLib/spdk `
NetworkPkg/Library/DxeSpdkLib/spdk should then be at `1216a672`

johnmeneghini committed:
Update spdk submodule for NBFT for failed cases

Update spdk submodule to point at latest timberland-1.0_final branch.
Without this update newly cloned repositories don't compile.

Signed-off-by: John Meneghini <jmeneghi@redhat.com>

johnmeneghini commented:
> > I'm having hard time getting this branch compiled, while other branches do compile fine for me:
> 
> This might be an issue with the cached spdk submodule not updating from remote. Can you give it a shot after reinitializing spdk with `--remote` after checking out? `git submodule update --remote NetworkPkg/Library/DxeSpdkLib/spdk` NetworkPkg/Library/DxeSpdkLib/spdk should then be at `1216a672`

I had the same issue. It's because you needed to update the git submodule in your branch. I've pushed a new commit to this branch that does that.  So if people retry they should be able to get this branch to completely compile.  Here are the steps I use:

```
git clone -b timberland_upstream-dev-full git@github.com:timberland-sig/edk2.git
pushd edk2
git config url."ssh://git@github.com/timberland-sig".insteadOf https://github.com/timberland-sig
git submodule update --init --recursive
make -C BaseTools clean
rm -rf Build
make -C BaseTools
source edksetup.sh
build -t GCC5 -a X64 -p OvmfPkg/OvmfPkgX64.dsc
```

tbzatek commented:
Confirming, this compiles fine now.

trevor-cockrell committed:
NetworkPkg/NvmeOfDxe: Add Controller ID support

Add Controller ID field to NVMeoF HII attempt pages.
Fix an issue with the Controller ID being incorrectly written to the
NBFT SSNS extended record.

Signed-off-by: Trevor cockrell <trevor_cockrell@dell.com>
Co-authored-by: Cole Harrington <cole.harrington@dell.com>

trevor-cockrell committed:
NetworkPkg/NvmeOfDxe: Resolve NBFT population issues (#35)

- Adjust HFI population logic to no longer skip by MAC, instead check
  for a unique transport descriptor hash
- Discovery ctlr population logic no longer treats each
  DeviceAdapter as an individual Discovery ctlr; Discovery ctlr entries
now populated based on order in which unique Disc. ctlrs were found
- Properly map SSNS, Discovery ctlrs to their respective primary HFI
- Properly map SSNS records to their respective primary Discovery ctlr

Signed-off-by: Trevor cockrell <trevor_cockrell@dell.com>

trevor-cockrell committed:
NetworkPkg/NvmeOfDxe: DHCP4 include Option61 (#30)

* NetworkPkg/NvmeOfDxe: DHCP4 include Option61

Provide DHCP Option61 (ClientID) to prevent OS receiving potential
address mismatch from that reported in NBFT, resulting in connection
failures.

Signed-off-by: Trevor cockrell <trevor_cockrell@dell.com>

* Bump copyrights

Signed-off-by: Trevor cockrell <trevor_cockrell@dell.com>

---------

Signed-off-by: Trevor cockrell <trevor_cockrell@dell.com>
Co-authored-by: Peter Liu <peter_liu@dell.com>

rahlvers commented:
Superceded by https://github.com/timberland-sig/edk2/pull/39
